docs/book/src/faq.md

```diff
@@ -0,0 +1,100 @@
+
+The domain is for the group suffix, to explicitly show the resource group category.
+For example, if set `--domain=example.com`:
+```
+kubebuilder init --domain example.com --repo xxx --plugins=go/v4-alpha
+kubebuilder create api --group mygroup --version v1beta1 --kind Mykind
+```
+Then the result resource group will be `mygroup.example.com`.
+
+> If domain field not set, the default value is `my.domain`.
+
+## I'd like to customize my project to use [klog][klog] instead of the [zap][zap] provided by controller-runtime. How to use `klog` or other loggers as the project logger?
+
+In the `main.go` you can replace:
+```go
+    opts := zap.Options{
+    Development: true,
+    }
+    opts.BindFlags(flag.CommandLine)
+    flag.Parse()
+    
+    ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+```
+with:
+```go
+    flag.Parse()
+	ctrl.SetLogger(klog.NewKlogr())
+```
+
+## After `make run`, I see errors like "unable to find leader election namespace: not running in-cluster..."
+
+You can enable the leader election. However, if you are testing the project locally using the `make run`
+target which will run the manager outside of the cluster then, you might also need to set the
+namespace the leader election resource will be created, as follows:
+```go
+mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:                  scheme,
+		MetricsBindAddress:      metricsAddr,
+		Port:                    9443,
+		HealthProbeBindAddress:  probeAddr,
+		LeaderElection:          enableLeaderElection,
+		LeaderElectionID:        "14be1926.testproject.org",
+		LeaderElectionNamespace: "<project-name>-system",
+```
+
+If you are running the project on the cluster with `make deploy` target
+then, you might not want to add this option. So, you might want to customize this behaviour using
+environment variables to only add this option for development purposes, such as:
+
+```go
+    leaderElectionNS := ""
+	if os.Getenv("ENABLE_LEADER_ELECATION_NAMESPACE") != "false" {
+		leaderElectionNS = "<project-name>-system"
+	}
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:                  scheme,
+		MetricsBindAddress:      metricsAddr,
+		Port:                    9443,
+		HealthProbeBindAddress:  probeAddr,
+		LeaderElection:          enableLeaderElection,
+		LeaderElectionNamespace: leaderElectionNS,
+		LeaderElectionID:        "14be1926.testproject.org",
+		...
+```
+
+## I am facing the error "open /var/run/secrets/kubernetes.io/serviceaccount/token: permission denied" when I deploy my project against Kubernetes versions <= 1.21. How to sort it out?
+
```

Comment:
See : https://github.com/kubernetes-sigs/kubebuilder/issues/3028#issuecomment-1293393346

- We cannot add the version if we are not 100% sure of what version it was solved (instead we can use old versions)
- We cannot add the fix PR if we are not 100% sure that it was the right one.

Reply:
Updated, PTAL thanks.